server/src/test/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenStoreTests.java

```diff
@@ -332,6 +337,32 @@ void beOAuth2StandardCompliant() {
         assertTrue(code.length() >= 32);
     }
 
+    @Test
+    void testCountingTheExecutedSqlDeleteStatements() throws SQLException {
+        // Given, mocked data source to count how often it is used, call performExpirationClean 10 times
+        DataSource mockedDataSource = mock(DataSource.class);
+        Connection mockedConnection = mock(Connection.class);
+        doReturn(mockedConnection).when(mockedDataSource).getConnection();
+        store = new UaaTokenStore(mockedDataSource);
+        try {
+            // warm up
+            store.performExpirationClean();
+        } catch (Exception sqlException) {
+            // expected during warm up
+        }
+        // When
+        for (int i = 0; i < 9; i++) {
+            // now start to cleanup in loop
+            try {
+                store.performExpirationClean();
+            } catch (Exception sqlException) {
+                // ignore
+            }
+        }
+        // Then
+        verify(mockedConnection, atMost(1)).prepareStatement("delete from oauth_code where expiresat > 0 AND expiresat < ?");
```

Comment:
One idea: it seems to me like we could for example change the signature of the `performExpirationClean` method, to return a boolean. It would be `true` if the cleanup happened, and `false` otherwise. 
That way, the assertions would be much easier to write, wouldn't they?